### PR TITLE
fix(train): handle error when draft PR is not supported by user

### PR DESCRIPTION
**Pull request by Syffe**
Some old plans of GitHub are not able to handle draft PR. We
needed to handle this case properly when train tries to create
a draft PR

Fixes MRGFY-1085

### CODE REVIEW

File: mergify_engine/queue/merge_train.py [hunk 1]
```diff
@@ -539,6 +539,11 @@ async def _create_draft_pull_request(
                 ).json(),
             )
         except http.HTTPClientSideError as e:
+
+            if "errors" not in e.response.json():
```
Comment (jd) [inline mergify_engine/queue/merge_train.py:543]: I find this cryptic. Can we add a comment at least?
Or even an example of the JSON you'd expect?

**Review by jd — DISMISSED** [review]

File: mergify_engine/queue/merge_train.py (hunk 1 above)
Comment (jd) [reply mergify_engine/queue/merge_train.py:543]: I'd even check for the message in the JSON to be clear.

Comment (Syffe) [reply mergify_engine/queue/merge_train.py:543]: Indeed referencing the expected json should do the trick

File: mergify_engine/queue/merge_train.py [hunk 2]
```diff
@@ -539,6 +539,11 @@ async def _create_draft_pull_request(
...
+                await self._set_creation_failure(e.response.json()["message"])
```
[the remark this answers is not in the document]
Comment (Syffe) [reply mergify_engine/queue/merge_train.py:544]: I'm not sure i quite understand the behavior, if the draft PR is not supported by user, we should trigger `raise tenacity.TryAgain`?

[the remark this answers is not in the document]
Comment (jd) [reply mergify_engine/queue/merge_train.py:544]: What he means is that you're doing a minimal version of what the code does in L547-582.
So it'd be better to rewrite L547-549 to include your condition here.

File: mergify_engine/clients/http.py [hunk 3]
```diff
@@ -64,12 +64,34 @@ class HTTPClientSideError(httpx.HTTPStatusError):
...
+            )
+
+        return (
+            True
+            if (
+                msg_to_find in self.response.json()["message"]
+                or msg_to_find in self.response.text
+            )
+            else False
+        )
```
Comment (sileht) [inline mergify_engine/clients/http.py:93]: It looks like I miss guided you, we don't use response.text anymore, so maybe it's safer to not handle this case

```suggestion
        return msg_to_find in self.response.json()["message"]
```

**Review by sileht — CHANGES_REQUESTED** [review]

File: mergify_engine/clients/http.py [hunk 4]
```diff
@@ -64,12 +64,27 @@ class HTTPClientSideError(httpx.HTTPStatusError):
...
 
+    def contains(self, msg_to_find: str) -> bool:
+
+        if "errors" in self.response.json():
+            return any(
+                msg_to_find in error["message"]
+                for error in self.response.json()["errors"]
+            )
+
+        return True if msg_to_find in self.response.json()["message"] else False
```
Comment (sileht) [inline mergify_engine/clients/http.py:86]:
```suggestion
        return msg_to_find in self.response.json()["message"]
```

**Review by sileht — CHANGES_REQUESTED** [review]

**Comment by sileht** [comment]
@mergify squash

**Review by sileht — APPROVED** [review]

**Pull request merged**